### PR TITLE
#162949038 User can login with social options

### PR DESCRIPTION
**Pull request by waracci**
#### What does this PR do?

Implement login with social options: Google, Facebook and Twitter

#### Description of Task to be completed?

- [x] setup firebase
- [x] setup Google Authentication
- [x] setup Twitter Authentication
- [x] setup Facebook Authentication

### Git clone the repository:

> `git clone https://github.com/andela/ah-robotics-frontend.git`

### Change the directory to the cloned repository:

> `cd ah-robotics-frontend`

### Check out to the feature branch

> `git checkout ft-social-login-162949038`

### Setting up the environment variables

Add a `.env` file with the following format.

> Kindly note that the secret key has been posted and pinned on the Slack channel.

```
   REACT_APP_API_KEY=<secret_key>
   REACT_APP_AUTH_DOMAIN=ah-robotics.firebaseapp.com
```

### Install required dependencies required to run the project

> `npm install`

### Start the server

> `npm start`

### Navigate to the login page on your browser

> `http://localhost:3000/login`

### Navigate to the review app to test the feature

><https://ah-robotics-frontend-sta-pr-14.herokuapp.com/>

### Running the test suite locally

> `npm test -- --coverage`

#### What are the relevant pivotal tracker stories?

[#162949038](https://www.pivotaltracker.com/story/show/162949038)

#### Screenshots (if appropriate)

1. Default login page

screenshot 2019-02-28 at 07 39 12

2. Click any of the providers and sign in with your credentials

screenshot 2019-02-28 at 07 39 34

3. Successful user login

screenshot 2019-02-28 at 07 40 35

### CODE REVIEW

**Review by johnwayodi — APPROVED** [review]
@waracci great work. I was able to login to the application via all social platforms.

**Review by Paulvitalis200 — COMMENTED** [review]
Hey @waracci,
I have managed to test this feature and it works well. Good work

**Comment by Teatoller** [comment]
Hi @waracci 
I have tested out the feature and the tests.
Feature working as instructed. 
The tests are all passing.

Regards
Steven Ennis

**Review by Teatoller — APPROVED** [review]
Hi @waracci
I have tested out the feature and the tests.
Feature working as instructed.
The tests are all passing.
Is it possible to have the Google icon pick up the red Google colors?
Regards
Steven Ennis

File: src/components/Article/article.scss [hunk 1]
```diff
@@ -27,8 +28,8 @@
   .article-body{
     color: black;
     margin-top: 30px;
-    letter-spacing: 1.4px;
-    font-family: Raleway, sans-serif;
+    // letter-spacing: 1.2px;
```
Comment (jomasim) [inline src/components/Article/article.scss]: remove the commented line

**Comment by jomasim** [comment]
Awesome work @waracci I have been to test the feature successfully.

File: src/components/Article/article.scss (hunk 1 above)
Comment (waracci) [reply src/components/Article/article.scss]: Okay, working on it.

**Pull request merged**